### PR TITLE
haskell.compiler.ghc8102Binary: fix w/ store path containing "xxx"

### PR DESCRIPTION
**Pull request by sternenseemann**
At some point in Makefile xxx is substituted for three spaces relating
to package files, causing it to access a nonexistent path.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
@ofborg build pkgsMusl.haskell.compiler.ghc8102Binary pkgsMusl.haskell.compiler.ghc8102BinaryMinimal

**Comment by sternenseemann** [comment]
@ofborg build haskell.compiler.ghc8102BinaryMinimal

**Comment by sternenseemann** [comment]
Filed issue https://gitlab.haskell.org/ghc/ghc/-/issues/21402

File: pkgs/development/compilers/ghc/8.10.2-binary.nix [hunk 1]
```diff
@@ -272,6 +272,12 @@ stdenv.mkDerivation rec {
     lib.optionalString stdenv.isLinux ''
       find . -type f -executable -exec patchelf \
           --interpreter ${stdenv.cc.bintools.dynamicLinker} {} \;
+    '' +
+    # The Makefile uses 'xxx' as a temporary placeholder in some path substitution
+    # machinery which can break the build if the store path / prefix happens to
+    # contain this string.
```
Comment (cdepillabout) [inline pkgs/development/compilers/ghc/8.10.2-binary.nix:278]:
```suggestion
    # contain this string.
    # https://gitlab.haskell.org/ghc/ghc/-/issues/21402
```

**Comment by cdepillabout** [comment]
@sternenseemann This fix seems reasonable to me.  Is there anything keeping you from merging this in?

**Comment by sternenseemann** [comment]
I want to add it for all versions that require it and in a more targeted manner (as it only affects the alpine bindists afaict.

**Comment by sternenseemann** [comment]
@ofborg build haskell.compiler.ghc8102Binary haskell.compiler.ghc8102BinaryMinimal haskell.compiler.ghc8107Binary haskell.compiler.ghc8107BinaryMinimal

**Comment by sternenseemann** [comment]
@ofborg build pkgsMusl.haskell.compiler.ghc8102Binary pkgsMusl.haskell.compiler.ghc8102BinaryMinimal pkgsMusl.haskell.compiler.ghc8107Binary pkgsMusl.haskell.compiler.ghc8107BinaryMinimal

**Review by cdepillabout — APPROVED** [review]
Looks good to me, thanks for taking the time to fix this up!

**Comment by cdepillabout** [comment]
We need to give 7 few days for `glirc` to be fixed from https://github.com/NixOS/nixpkgs/pull/169513#issuecomment-1106558920, so I imagine now is a pretty good time to do the set rebuild.

**Pull request merged**